app/models/context.rb

```diff
@@ -7,4 +8,8 @@
   def system_settings
     self[:system_settings] ||= SystemSetting.current_settings
   end
+
+  def can_admin?
+    user && (user&.admin_role? || user&.sys_admin_role?) && admin_param != 'false'
+  end
```

Comment:
I think I disagree with this. I'm fine with the PR going ahead with it, though, since I think it will be easy to edit later. 

Here's my thinking. I read your comment in the commit that says

> While this duplicates a bit of logic in ApplicationPolicy, it didn't
feel quite right to have to do `policy(model).change?` in a
controller to determine where to redirect to. The `.change?` part of
that call seems an arbitrary choice. In fact, while it seems related to
permissions, in this case what i'm asking is not whether the user has
permissions to do something but whether they are acting as an admin.

and compared that with how you use this method in https://github.com/rubyforgood/mutual-aid/blob/a6dac92e081ed3e5bd69b7e17400b4df49bf8b58/app/controllers/announcements_controller.rb#L50-L56

I can imagine a scenario where we want to do something like — "allow sys_admins to approve announcements but not regular admins"

If we use the policy(...).change? check in the controller sample I linked to in here, then we'd only need to set that logic in the policy, and the controller would automatically follow the new behavior. Whereas if we use `context.can_admin?` now we need to edit something — either `context.can_admin?` or something in the controller — to allow this new behavior.

Of course, this is a hypothetical change and I'm happy to worry about it later when we get there. So I'm fine with this being merged in even though what you call an arbitrary choice feels less arbitrary to me. I think this is probably influenced by my background in administering weird Windows Active Directory permission environments.

Reply:
Great points, i like all of that thinking. Pushing this logic into the policy, maybe with a custom query like `.review?` would make a lot of sense and be more future-proof as you describe. I think i was influenced by realizing several controllers need this kind of logic and so we'd need similar methods in several policies whereas `context.can_admin?` can be reused.

All that said, i think this is well worth keeping an eye on. As with most things, we'll learn as we use it more what works and what doesn't.

Appreciate the rigorous look! 💥.